Modules/xSharePoint/DSCResources/MSFT_xSPWebApplicationAppDomain/MSFT_xSPWebApplicationAppDomain.psm1

```diff
@@ -0,0 +1,124 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [parameter(Mandatory = $true)]  [System.String] $AppDomain,
+        [parameter(Mandatory = $true)] [System.String] $Prefix,
+        [parameter(Mandatory = $false)]  [System.String] $WebApplication,
+        [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount,
+        [parameter(Mandatory = $false)] [System.String] $Zone,
+        [parameter(Mandatory = $false)] [System.UInt32] $Port,
+        [parameter(Mandatory = $false)] [System.Boolean] $SSL
+
+    )
+
+    Write-Verbose -Message "Checking app urls settings"
+
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+        $params = $args[0]
+        
+        $prefix = Get-SPAppSiteSubscriptionName -ErrorAction SilentlyContinue
+        if($params.ContainsKey("WebApplication")){
+            $params.Add("Confirm", $false)
+            $ssl = $params | Remove-xSharePointParamValue -name "SSL" 
+            $appDomain = $params | Remove-xSharePointParamValue -name "AppDomain"
+            $port = $params | Remove-xSharePointParamValue -name "Port"
+            $webAppAppDomain =   Get-SPWebApplicationAppDomain $params
+            return @{
+                AppDomain = $webAppAppDomain.AppDomain 
+                WebApplication = $webAppAppDomain.WebApplication
```

Comment:
Does the WebApplication property here get returned as a string or SPWebApplication object?